Mathlib/Data/Real/EReal.lean

```diff
@@ -832,6 +865,20 @@ theorem add_lt_top {x y : EReal} (hx : x ≠ ⊤) (hy : y ≠ ⊤) : x + y < ⊤
   rw [← EReal.top_add_top]
   exact EReal.add_lt_add hx.lt_top hy.lt_top
 
+lemma add_ne_top {x y : EReal} (hx : x ≠ ⊤) (hy : y ≠ ⊤) : x + y ≠ ⊤ := by
+  rw [← lt_top_iff_ne_top]
+  exact add_lt_top hx hy
+
+lemma add_ne_top_iff_of_ne_bot {x y : EReal} (hx : x ≠ ⊥) (hy : y ≠ ⊥) :
```

Comment:
I don't really understand the name of this lemma

Reply:
I'm not sure how else we could call it, this is basically the description of the statement: if the elements are not bot (`of_ne_bot`), then the sum is not top (`add_ne_top`) if and only if (`iff`) something...
The only other name that comes to mind is `add_ne_top_iff_ne_top_of_ne_bot`, which is even longer.

Reply:
Well, but the `of_` part refers to the assumptions `hx` and `hy`. I would go for `add_ne_top_iff_ne_top2` (with the `2` as a subscript, but cannot do this in this github comment...)